pkg/webhook/nad/validator.go

```diff
@@ -126,18 +126,18 @@ func (v *Validator) checkRoute(config string) error {
 }
 
 func (v *Validator) checkVmi(nad *cniv1.NetworkAttachmentDefinition) error {
```

Comment:
This fix, will break the `update nad` feature, as `checkVmi` is also used in `func (v *Validator) Update(_ *admission.Request, oldObj, newObj runtime.Object)`.

When user change nad, all VMIs need to be stopped. If it is storagenetwork nad, user must clear storagenetwork/set-to-another storagenetwork nad first.

When user delete nad, all VMs need to be changed to remove the network referrences. If it is storagenetwork nad, user must clear storagenetwork/set-to-another storagenetwork nad first.

A new function `checkVM` is needed for nad deletion webhook.

Other changes need to reviewed similarly.

Reply:
@rrajendran17  Change the `Vlan,HairpinMode,PromiscMode and some others`  of the NAD params is reasonable, stop & start VM is okay;  it is not a good idea to force user to delete the VM/VM's related interface.


```
type NetConf struct {
	cniv1.NetConf
	BrName       string `json:"bridge"`
	IsGW         bool   `json:"isGateway"`
	IsDefaultGW  bool   `json:"isDefaultGateway"`
	ForceAddress bool   `json:"forceAddress"`
	IPMasq       bool   `json:"ipMasq"`
	MTU          int    `json:"mtu"`
	HairpinMode  bool   `json:"hairpinMode"`
	PromiscMode  bool   `json:"promiscMode"`
	Vlan         int    `json:"vlan"`
}
```


change `BrName` (`clusternetwork`) is rare or should be denied.